toolchain/parser/parse_node_kind.def

```diff
@@ -9,78 +9,99 @@
 // desired output. The x-macro for this header is `CARBON_PARSE_NODE_KIND`. The
 // definition provided will be removed at the end of this file to clean up.
 
-#ifndef CARBON_PARSE_NODE_KIND
-#error "Must define the x-macro to use this file."
+#if !(defined(CARBON_PARSE_NODE_KIND) ||          \
+      (defined(CARBON_PARSE_NODE_KIND_BRACKET) && \
+       defined(CARBON_PARSE_NODE_KIND_CHILD_COUNT)))
+#error "Must define the CARBON_PARSE_NODE_KIND x-macro to use this file."
 #endif
 
+// The BRACKET and CHILD_COUNT macros will use CARBON_PARSE_NODE_KIND by default
+// when undefined.
+#ifndef CARBON_PARSE_NODE_KIND_BRACKET
+#define CARBON_PARSE_NODE_KIND_BRACKET(Name, ...) CARBON_PARSE_NODE_KIND(Name)
+#endif
+#ifndef CARBON_PARSE_NODE_KIND_CHILD_COUNT
+#define CARBON_PARSE_NODE_KIND_CHILD_COUNT(Name, ...) \
+  CARBON_PARSE_NODE_KIND(Name)
+#endif
```

Comment:
Would be good to document how these are expected to be used below, what the semantics of the parameters are, etc. I think I can guess from reading, but seems good to be explicit in a comment.

Reply:
Added comments to file